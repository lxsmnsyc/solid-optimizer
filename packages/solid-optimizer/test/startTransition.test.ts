import { describe, expect, it } from 'vitest';
import compile from './compile';

describe('startTransition', () => {
  describe('callbacks w/o body', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { startTransition } from 'solid-js';
        
        startTransition(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { startTransition as x } from 'solid-js';
        
        x(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'startTransition' as x } from 'solid-js';
        
        x(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.startTransition(() => update());
      `)).toMatchSnapshot();
    });
  });
  describe('callbacks with body', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { startTransition } from 'solid-js';
        
        startTransition(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { startTransition as x } from 'solid-js';
        
        x(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'startTransition' as x } from 'solid-js';
        
        x(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.startTransition(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
  });
  describe('spreads', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { startTransition } from 'solid-js';
        
        startTransition(...example);
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { startTransition as x } from 'solid-js';
        
        x(...example);
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'startTransition' as x } from 'solid-js';
        
        x(...example);
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.startTransition(...example);
      `)).toMatchSnapshot();
    });
  });
});
