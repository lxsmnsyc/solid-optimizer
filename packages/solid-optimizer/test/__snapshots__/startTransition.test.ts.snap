// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`startTransition > callbacks w/o body > should work for ImportSpecifier 1`] = `
"import { startTransition } from 'solid-js';
update();"
`;

exports[`startTransition > callbacks w/o body > should work for aliased ImportSpecifier 1`] = `
"import { startTransition as x } from 'solid-js';
update();"
`;

exports[`startTransition > callbacks w/o body > should work for aliased string ImportSpecifier 1`] = `
"import { 'startTransition' as x } from 'solid-js';
update();"
`;

exports[`startTransition > callbacks w/o body > should work for namespace 1`] = `
"import * as solid from 'solid-js';
update();"
`;

exports[`startTransition > callbacks with body > should work for ImportSpecifier 1`] = `
"import { startTransition } from 'solid-js';
(() => {
  update();
})();"
`;

exports[`startTransition > callbacks with body > should work for aliased ImportSpecifier 1`] = `
"import { startTransition as x } from 'solid-js';
(() => {
  update();
})();"
`;

exports[`startTransition > callbacks with body > should work for aliased string ImportSpecifier 1`] = `
"import { 'startTransition' as x } from 'solid-js';
(() => {
  update();
})();"
`;

exports[`startTransition > callbacks with body > should work for namespace 1`] = `
"import * as solid from 'solid-js';
(() => {
  update();
})();"
`;

exports[`startTransition > spreads > should work for ImportSpecifier 1`] = `
"import { startTransition } from 'solid-js';
example[0]();"
`;

exports[`startTransition > spreads > should work for aliased ImportSpecifier 1`] = `
"import { startTransition as x } from 'solid-js';
example[0]();"
`;

exports[`startTransition > spreads > should work for aliased string ImportSpecifier 1`] = `
"import { 'startTransition' as x } from 'solid-js';
example[0]();"
`;

exports[`startTransition > spreads > should work for namespace 1`] = `
"import * as solid from 'solid-js';
example[0]();"
`;
